sdk/identity/azure-identity/test/ut/managed_identity_credential_test.cpp

```diff
@@ -26,2166 +26,1510 @@
+    EXPECT_EQ(request2.Body, std::string());
+
+    {
+      EXPECT_NE(request0.Headers.find("Metadata"), request0.Headers.end());
+      EXPECT_EQ(request0.Headers.at("Metadata"), "true");
+
+      EXPECT_NE(request1.Headers.find("Metadata"), request1.Headers.end());
+      EXPECT_EQ(request1.Headers.at("Metadata"), "true");
+
+      EXPECT_NE(request2.Headers.find("Metadata"), request2.Headers.end());
+      EXPECT_EQ(request2.Headers.at("Metadata"), "true");
+    }
+
+    EXPECT_EQ(response0.AccessToken.Token, "ACCESSTOKEN1");
+    EXPECT_EQ(response1.AccessToken.Token, "ACCESSTOKEN2");
+    EXPECT_EQ(response2.AccessToken.Token, "ACCESSTOKEN3");
+
+    using namespace std::chrono_literals;
+    EXPECT_GE(response0.AccessToken.ExpiresOn, response0.EarliestExpiration + 3600s);
+    EXPECT_LE(response0.AccessToken.ExpiresOn, response0.LatestExpiration + 3600s);
+
+    EXPECT_GE(response1.AccessToken.ExpiresOn, response1.EarliestExpiration + 3600s);
+    EXPECT_LE(response1.AccessToken.ExpiresOn, response1.LatestExpiration + 3600s);
+
+    EXPECT_GE(response2.AccessToken.ExpiresOn, response2.EarliestExpiration + 4999s);
+    EXPECT_LE(response2.AccessToken.ExpiresOn, response2.LatestExpiration + 4999s);
+
+    Logger::SetListener(nullptr);
+  }
+
+  TEST(ManagedIdentityCredential, CloudShellClientId)
+  {
+    using Azure::Core::Credentials::AuthenticationException;
+
+    static_cast<void>(CredentialTestHelper::SimulateTokenRequest(
+        [](auto transport) {
+          TokenCredentialOptions options;
+          options.Transport.Transport = transport;
+
+          CredentialTestHelper::EnvironmentOverride const env({
+              {"MSI_ENDPOINT", "https://microsoft.com/"},
+              {"MSI_SECRET", ""},
+              {"IDENTITY_ENDPOINT", "https://visualstudio.com/"},
+              {"IMDS_ENDPOINT", "https://xbox.com/"},
+              {"IDENTITY_HEADER", ""},
+              {"IDENTITY_SERVER_THUMBPRINT", "0123456789abcdef0123456789abcdef01234567"},
+          });
+
+          std::unique_ptr<ManagedIdentityCredential const> cloudShellManagedIdentityCredential;
+          EXPECT_THROW(
+              cloudShellManagedIdentityCredential = std::make_unique<ManagedIdentityCredential>(
+                  "fedcba98-7654-3210-0123-456789abcdef", options),
+              AuthenticationException);
+
+          return cloudShellManagedIdentityCredential;
+        },
+        {},
+        {"{\"expires_in\":3600, \"access_token\":\"ACCESSTOKEN1\"}"}));
+  }
+
+  TEST(ManagedIdentityCredential, CloudShellResourceId)
+  {
+    using Azure::Core::Credentials::AuthenticationException;
+
+    static_cast<void>(CredentialTestHelper::SimulateTokenRequest(
+        [](auto transport) {
+          ManagedIdentityCredentialOptions options;
+          options.Transport.Transport = transport;
+          options.IdentityId = ManagedIdentityId::FromUserAssignedResourceId(
+              ResourceIdentifier("abcdef01-2345-6789-9876-543210fedcba"));
```

Comment:
I'd expect the resource ID to not be a GUID and instead follow this format: https://github.com/Azure/azure-sdk-for-net/blob/c5bc53adb7006da46c9e5cbe1092f6e1e074b76d/sdk/identity/Azure.Identity/tests/ManagedIdentityCredentialTests.cs#L29

Reply:
Agreed.
In this case, it's just a mocked test, so we are mainly checking that any string we provide ends up in the right place within the request. The semantics of the string don't matter for the test.

I'll update it to match more closely to a resource ID format, once we add input validation to the ResourceIdentifier type.

Reply:
My thinking here is that more experienced devs will look at the tests instead of the docs. For that reason, I think it's important to bring the tests closer to reality.